core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt

```diff
@@ -219,4 +221,77 @@ class LockFile {
     var lockData = UUID.randomUUID().toString()
 }
 
-class DropBoxFileConflictException: Exception()
\ No newline at end of file
+/**
+ *	Wrapper around OnlineMultiplayer's synchronization facilities.
+ *
+ *	Based on the design of Mutex from kotlinx.coroutines.sync, except that when it blocks,
+ *	  it blocks the entire thread for an increasing period of time via Thread.sleep()
+ */
+class ServerMutex(val gameInfo: GameInfo) {
+	var locked = false
+
+    /**
+     * Try to obtain the server lock ONCE
+     * @see OnlineMultiplayer.tryLockGame
+     * @see lock
+     * @see unlock
+     * @return true if lock is acquired
+     */
+	fun tryLock(): Boolean {
+		locked = OnlineMultiplayer().tryLockGame(gameInfo.asPreview())
+        return locked
+	}
+
+    /**
+     * Block until this client owns the lock
+     *
+     * TODO: Create an alternative to the underlying tryLock or tryLockGame which checks for
+     *       (and returns, when present) the value of the Retry-After header
+     *
+     * @see tryLock
+     * @see unlock
+     */
+	fun lock() {
+		var tries = 0
+
+        // Try the lockfile once
+        locked = tryLock()
+		while (!locked) {
+            // Wait exponentially longer after each attempt as per DropBox API recommendations
+            var delay = 250 * 2.0.pow(tries).toLong()
+
+            // 8 seconds is a really long time to sleep a thread, it's as good a cap as any
+            if (delay > 8000) {
+                delay = 8000
+            }
+
+            // Consider NOT sleeping here, instead perhaps delay or spin+yield
+			Thread.sleep(delay)
+
+            tries++
+
+            // Retry the lock
+            locked = tryLock()
+		}
```

Comment:
The indentation looks kinda wonky and should probably be adjusted for readability. Seems like there are sometimes tabs, sometimes two spaces, and sometimes more? Four spaces is what the rest of the code is going with (I think four spaces is also the default Kotlin setting in Android Studio)

Reply:
Oops... Sometimes I make tweaks before a commit using vim; it probably threw in a few tabs on me. Android Studio renders those correctly, so I never noticed. I'll fix them.